File: src/main/java/gregtechlite/gtlitecore/core/command/CommandManagerImpl.java
package gregtechlite.gtlitecore.core.command;

import gregtechlite.gtlitecore.api.command.CommandManager;
import net.minecraft.command.ICommand;
import net.minecraftforge.fml.common.event.FMLServerStartingEvent;
import net.minecraftforge.server.command.CommandTreeBase;
import org.jetbrains.annotations.ApiStatus.Internal;

@Internal
public class CommandManagerImpl implements CommandManager
{

    private static final CommandManagerImpl INSTANCE = new CommandManagerImpl();
    private final CommandTreeBase command = new Commands();

    private CommandManagerImpl() {}

    public static CommandManagerImpl getInstance()
    {
        return INSTANCE;
    }

    @Override
    public void addCommand(ICommand command)
    {
        this.command.addSubcommand(command);
    }

    public void registerServerCommand(FMLServerStartingEvent event)
    {
        event.registerServerCommand(this.command);
    }

}

File: src/main/kotlin/gregtechlite/gtlitecore/core/command/CommandManagerImpl.kt
package gregtechlite.gtlitecore.core.command

import gregtechlite.gtlitecore.api.command.CommandManager
import net.minecraft.command.ICommand
import net.minecraftforge.fml.common.event.FMLServerStartingEvent
import net.minecraftforge.server.command.CommandTreeBase

class CommandManagerImpl : CommandManager
{
    private val command: CommandTreeBase = Commands()

    companion object
    {
        private val instance = CommandManagerImpl()

        fun getInstance(): CommandManagerImpl = instance
    }

    override fun addCommand(command: ICommand)
    {
        this.command.addSubcommand(command)
    }

    fun registerServerCommand(event: FMLServerStartingEvent)
    {
        event.registerServerCommand(command)
    }

}